src/widgets-service.js

```diff
@@ -31,25 +30,30 @@ WidgetsService.prototype.createCategoryModel = function (attrs, layer) {
   try {
     _checkProperties(attrs, ['title', 'column']);
   } catch (err) {
-    cdb.log.error('Error creating newCategoryModel, ' + err.message);
-    return;
+    throw new Error('Error creating newCategoryModel, ' + err.message);
   }
 
-  var dataviewModel = this._dataviews.createCategoryModel(layer, {
+  var dataviewModel = this._dataviews.createCategoryModel(layer, this._dataviewModelAttrs(attrs, {
     type: 'category',
     column: attrs.column,
     aggregation: attrs.aggregation || 'count',
     aggregation_column: attrs.aggregation_column || attrs.column,
-    suffix: attrs.suffix,
-    prefix: attrs.prefix
-  });
+    prefix: attrs.prefix,
+    suffix: attrs.suffix
+  }));
 
   var widgetModel = new CategoryWidgetModel({
     id: attrs.id,
     type: 'category',
     title: attrs.title,
     attrsNames: ['title', 'collapsed'],
-    dataviewModelAttrsNames: ['column', 'aggregation', 'aggregation_column']
+    dataviewModelAttrsNames: this._dataviewModelAttrsNames([
```

Comment:
We could even use a constant, since these are actually constant:

```
DEFAULT_DATAVIEW_MODEL_ATTRIBUTES = [
  'sync_on_data_change',
  'sync_on_bbox_change',
  'enabled'
]
```